src/nrfx/hal/nrf_rng.h

```diff
@@ -90,98 +90,109 @@ typedef enum
  */
-static inline bool nrf_rng_int_get(nrf_rng_int_mask_t rng_int_mask);
+static inline uint32_t nrf_rng_int_enable_check(NRF_RNG_Type const * p_reg, uint32_t mask);
 
 /**
- * @brief Function for setting a specific task.
+ * @brief Function for triggering the specified task.
  *
- * @param[in]  rng_task              Task.
+ * @param[in] p_reg    Pointer to the structure of registers of the peripheral.
+ * @param[in] rng_task The specified Task.
  */
-void nrf_rng_task_trigger(nrf_rng_task_t rng_task);
+void nrf_rng_task_trigger(NRF_RNG_Type * p_reg, nrf_rng_task_t rng_task);
 
 /**
- * @brief Function for clearing a specific event.
+ * @brief Function for clearing the specified event.
  *
- * @param[in]  rng_event              Event.
+ * @param[in] p_reg     Pointer to the structure of registers of the peripheral.
+ * @param[in] rng_event The specified event.
  */
-static inline void nrf_rng_event_clear(nrf_rng_event_t rng_event);
+static inline void nrf_rng_event_clear(NRF_RNG_Type * p_reg, nrf_rng_event_t rng_event);
 
 /**
- * @brief Function for getting the state of a specific event.
+ * @brief Function for retrieving the state of the specified event.
  *
- * @param[in]  rng_event              Event.
+ * @param[in] p_reg     Pointer to the structure of registers of the peripheral.
+ * @param[in] rng_event The specified event.
  *
- * @retval     true               If the event is not set.
- * @retval     false              If the event is set.
+ * @retval true  The event is set.
+ * @retval false The event is not set.
  */
-static inline bool nrf_rng_event_get(nrf_rng_event_t rng_event);
+static inline bool nrf_rng_event_check(NRF_RNG_Type const * p_reg, nrf_rng_event_t rng_event);
 
 /**
  * @brief Function for getting the previously generated random value.
  *
- * @return     Previously generated random value.
+ * @param[in] p_reg Pointer to the structure of registers of the peripheral.
+ *
+ * @return Previously generated random value.
  */
-static inline uint8_t nrf_rng_random_value_get(void);
+static inline uint8_t nrf_rng_random_value_get(NRF_RNG_Type * p_reg);
 
 /**
  * @brief Function for enabling digital error correction.
+ *
+ * @param[in] p_reg Pointer to the structure of registers of the peripheral.
  */
-static inline void nrf_rng_error_correction_enable(void);
+static inline void nrf_rng_error_correction_enable(NRF_RNG_Type * p_reg);
 
 /**
  * @brief Function for disabling digital error correction.
+ *
+ * @param[in] p_reg Pointer to the structure of registers of the peripheral.
  */
-static inline void nrf_rng_error_correction_disable(void);
+static inline void nrf_rng_error_correction_disable(NRF_RNG_Type * p_reg);
 
 
 
 /* Bodies for inlined functions  */
 
-static inline bool nrf_rng_int_get(nrf_rng_int_mask_t rng_int_mask)
+static inline uint32_t nrf_rng_int_enable_check(NRF_RNG_Type const * p_reg, uint32_t mask)
 {
-    return (bool)(NRF_RNG->INTENSET & rng_int_mask);
+    return p_reg->INTENSET & mask;
```

Comment:
why p_reg is used here when in other function NRF_RNG is used?

Reply:
Because for this one I needed to also update the implementation with the real nrfx HAL one, and I apparently forgot to use `NRF_RNG` as in other functions. Actually, this makes no difference, since there is only one instance of `NRF_RNG`, but for consistency I'll align this function with others.
Thanks for catching this.

Reply:
Done.